src/MongoDB.Driver/Linq/Processors/AccumulatorBinder.cs

```diff
@@ -198,6 +198,19 @@ private bool TryGetAccumulatorTypeAndArgument(PipelineExpression node, out Accum
             return false;
         }
 
+        private Expression GetCountAccumulatorArgument(Expression node)
+        {
+            var where = node as WhereExpression;
+            if (where != null)
+            {
+                return Expression.IfThenElse(where.Predicate, Expression.Constant(1), Expression.Constant(0));
+            }
+            else
+            {
+                return Expression.Constant(1);
```

Comment:
Sure... I understand that this is just doing what we were doing before when the node is not a WhereExpression.

My question is why that is correct?

Seems like we should check that node is something for which 1 is correct, and throw some sort of exception if not supported, but I don't understand what's happening here well enough to be more concrete.

Reply:
As I understand, here we are generating a parameter for `$sum` operation. The parameter for this operation determines which records should be summarized to calculate the `Count` result. 

If a calculated parameter for some record has the `1` value, it will mean this record should be used in the calculation of `Count`, otherwise - nope.

In case of not `WhereExpression` when we have only `Expression.Constant(1)`, it means the same as having a `Count` method without predicate (just Count() ):

> My question is why that is correct?
which expression can we have here except of `WhereExpression`?
- In case if we have empty Count predicate, the `node` variable will have `DocumentExpression` type which is correct if we use just `Expression.Constant(1)`

Reply:
I don't see which else parameter can be pushed here through `Count` predicate. So, I'm not sure that throwing exception will help here.

Reply:
Maybe @craiggwilson needs to chime in here... I'm just saying that we're ignoring the node parameter without any sort of verification of what it might be and blindly(?) assuming that Expression.Constant(1) is the right thing to do.

I understand that this is what we were doing before... if that's a good enough reason then OK.

Reply:
We only get here when the result operator is a CountResultOperator. If we look at the overloads of the LINQ Count methods, there is one with no argument, which means `1`, and one with a predicate, which is what this PR is accounting for.